Production environment docs

brockfanning opened the pull request:
Fixes #536

brockfanning commented:
@LucyGwilliamAdmin This is my first draft of the production environment documentation. In the course of writing this up, I added some Github Actions workflow files to the three start repos:
* https://github.com/open-sdg/open-sdg-site-starter/blob/develop/.github/workflows/deploy-to-production.yml
* https://github.com/open-sdg/open-sdg-data-starter/blob/develop/.github/workflows/deploy-to-production.yml
* https://github.com/open-sdg/open-sdg-simple-starter/blob/develop/.github/workflows/deploy-to-production.yml

If you have time, could you revise the docs as needed and maybe run through the steps to make sure it's correct?

LucyGwilliamAdmin commented:
@brockfanning I think we need to add something about creating a master branch - is this just done in the normal way of creating a branch or are there any extra steps because it's master?

LucyGwilliamAdmin commented:
@brockfanning 
Further update: Also tried manually pushing to master in live site repo and this also didn't make the [website](https://open-sdg-simple-starter-test.github.io/) build

LucyGwilliamAdmin commented:
Actually "Deploy to production is being triggered" - I was expecting it to show in the PR but I can see it when I go to the Actions page

LucyGwilliamAdmin commented:
I just changed the branch from gh-pages to master [here](https://github.com/LucyGwilliamAdmin/simple-starter-test/blob/develop/.github/workflows/deploy-to-production.yml#L49) and now my site is building

LucyGwilliamAdmin commented:
@brockfanning unfortunately I don't think I'm unable to revise the docs as I'm not really sure what needs to be changed. I'm sort of just using trial and error.

brockfanning commented:
@LucyGwilliamAdmin I finally got this updated. I think it should work now. A couple of critical changes were:
1. Apparently the initial deploy doesn't work on a completely empty repository. So I added instructions to check the "Initialize with README" box when creating the production repos.
2. I had the production site deploying to gh-pages, but actually it should go to master.
3. I added instructions on creating the master branches, like you suggest.

The docs in this PR and the starter repos should be fixed now, in case you have a chance to try again.

LucyGwilliamAdmin commented:
@brockfanning just trying to test this before tech call.

Going from beginning on quick start just to have final check of whole process but I seem to getting an error:
https://github.com/LucyGwilliamAdmin/sdg-site/pull/3/checks?check_run_id=570927790

LucyGwilliamAdmin commented on docs/hosting/github-pages-production.md:
```diff
@@ -0,0 +1,103 @@
+
+Just as with the data repository, to perform the *first* production deployment of your **site**, you need to create a `master` branch. As before, this can be done as follows:
+
+1. Go to your data repository in GitHub.com.
```
Should this be "Go to your **site** repository in GitHub.com

LucyGwilliamAdmin commented on docs/deployment.md:
```diff
@@ -21,7 +21,7 @@ However, alternatives are easy - after all, it's merely uploading files.
 ## Staging vs Production
 
-It is helpful to have both a "staging" and "production" version of the site, in order to preview changes (staging) before they are presented to the public (production). If using Github Pages as a host, the simplest way to do this to create a second Github account specifically for the purpose of hosting the production site.
+It is helpful to have both a "staging" and "production" version of the site, in order to preview changes (staging) before they are presented to the public (production). More details on this can be found [here](./hosting/production.md).
```
"here" link should be more descriptive

LucyGwilliamAdmin commented:
@brockfanning great, I was able to set up a prod site using the latest docs!

Was thinking - should we do the same here i.e. trigger rebuild of prod site when prod data is built?

brockfanning commented:
@LucyGwilliamAdmin Good points, this is ready for review again.
